### PR TITLE
[Core] Determinants of parents in quadrature point geometries

### PR DESCRIPTION
**Pull request by tteschemacher**
Dear all,

this is another beautiful PR, which might be seen by some, checked by few and ignored by many. It allows to proceed with a certain flexibility and brings in a significant possibility of improving the efficiency. So it invites fruitful and amazing discussion. Nevertheless, it requires a set of previous PRs to be merged. Specifically mentioning, #7491, #7552. 

**Description**
Adding access to the determinants of the jacobians of the parents of the quadrature point geometries. This has the advantage that the quad point geometries can be stored with a reduced number of points, if they are non-zero. Nevertheless, for derivatives, other nodes might have non-zero values. As only the jacobian/ determinant is required this should just be passed to the parent.

**Changelog**
So far so sweet. Thanks for letting me have another permanent part of the PR list ;).

### CODE REVIEW

File: kratos/geometries/quadrature_point_geometry.h [hunk 1]
```diff
@@ -238,6 +238,21 @@ class QuadraturePointGeometry
...
+    ///@}
+    ///@name Dynamic access to internals
+    ///@{
+
+    /// Calculate with Vector
+    void Calculate(
+        const Variable<Vector>& rVariable,
+        Vector& rOutput) override
+    {
+        if (rVariable == DETERMINANTS_OF_JACOBIAN_PARENT)
```
Comment (loumalouomega) [inline kratos/geometries/quadrature_point_geometry.h]:
```suggestion
        if (rVariable == DETERMINANTS_OF_JACOBIAN_PARENT) {
```

File: kratos/geometries/quadrature_point_geometry.h [hunk 2]
```diff
@@ -238,6 +238,21 @@ class QuadraturePointGeometry
...
+        {
```
Comment (loumalouomega) [inline kratos/geometries/quadrature_point_geometry.h]:
```suggestion
```

File: kratos/geometries/geometry.h [hunk 3]
```diff
@@ -634,6 +634,99 @@ class Geometry
...
+    ///@}
+    ///@name Dynamic access to internals
+    ///@{
+
+    /* Set s a certain value to the geometry,
+     * according to a variable.
+     * Allows dynamic interfaces with each respective geometry.
+     */
+
+    /// Set with bool
```
Comment (loumalouomega) [inline kratos/geometries/geometry.h]: Can you add amore extended datailed Doxygen?

**Review by loumalouomega — COMMENTED** [review]
Must be approved by the @KratosMultiphysics/technical-committee

**Review by philbucher — APPROVED** [review]

**Pull request merged**